server.coffee

```diff
@@ -136,14 +148,21 @@ extendPublish (name, publishFunction, options) ->
       # This can hide some errors in publish functions if they one calls "added" on an existing document and we could
       # make it so that this behavior works only inside reactive computation (if "currentComputation" is set), but we
       # can also make it so that publish function tries to do something smarter (sending a diff) in all cases, as we do.
-      if @_documents[collectionName]?[stringId]
+      if ((@_documents instanceof Map && @_documents.get(collectionName)?.has(stringId)) || @_documents[collectionName]?[stringId])
         oldFields = {}
         # If some field existed before, but does not exist anymore, we have to remove it by calling "changed"
         # with value set to "undefined". So we look into current session's state and see which fields are currently
         # known and create an object of same fields, just all values set to "undefined". We then override some fields
         # with new values. Only top-level fields matter.
-        for field of @_session.getCollectionView(collectionName)?.documents?[stringId]?.dataByKey or {}
+        _documents = @_session.getCollectionView(collectionName)?.documents or {}
+        if _documents instanceof Map
+          dataByKey = _documents.get(stringId)?.dataByKey or {}
+        else
+          dataByKey = _documents?[stringId]?.dataByKey or {}
+
+        iterateObjectOrMapKeys dataByKey, (field) =>
           oldFields[field] = undefined
```

Comment:
Are you sure in Meteor 1.8 they are still setting it to undefined?

Reply:
Yes, that did not change in Meteor 1.8.1.